### PR TITLE
Bug 1309547 - Add new atmo.keys application

### PR DESCRIPTION
**Pull request by maurodoglio**
This will also migrate the existing public keys from the clusters app
to the new model.

In case there is no SSH key available users will be asked to create
one first.

The bulk of the changes are in 29b67349cba1ff60262285df13a054d6c2753af9.

### CODE REVIEW

**Comment by maurodoglio** [comment]
@jezdez this is what I have done so far. There are a couple of things still missing before we can consider this complete: integration with the cluster creation form and tests for the public-key-list view. Let me know if you prefer to pick it up from here or not.

**Comment by jezdez** [comment]
@maurodoglio this is now ready for review, check out the following screenshots:

screen-shot-2017-01-16-16-53-45

screen-shot-2017-01-16-16-52-29

**Comment by jezdez** [comment]
@maurodoglio Mind you, this includes the changes in #86 (dockerflow update) since I didn't want to go back again. Let's wait for that PR to merge before merge this one.

File: atmo/clusters/views.py [hunk 1]
```diff
@@ -13,6 +15,16 @@
 
 @login_required
 def new_cluster(request):
+    if request.user.created_sshkeys.count() == 0:
+        messages.error(
+            request,
+            mark_safe(
+                '<h4>No SSH keys found for you in the database.</h4>'
+                'Please upload one below to be able to launch a cluster.'
+                'This is one-time-step.'
```
Comment (robhudson) [inline atmo/clusters/views.py]: "one-time step"

File: atmo/keys/views.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+@delete_permission_required(SSHKey)
+def delete_key(request, id):
+    key = get_object_or_404(SSHKey, pk=id)
+    last_key = key.created_by.created_sshkeys.count() == 1
+    if last_key:
+        messages.error(
+            request,
+            mark_safe(
+                '<h4>Key not deleted!</h4> '
+                'At least one SSH key needs to exists, please add '
```
Comment (robhudson) [inline atmo/keys/views.py]: "needs to exist" sounds more correct to me.

File: atmo/static/js/keys.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      var reader = new FileReader();
+      // Closure to capture the file information.
+      reader.onload = function(event) {
+        if (!title_input.val()) {
+          title_input.val(file.name.replace(/.pub/g, ''));
+        };
+        key_textarea.val(event.target.result);
+      };
+
+      // Read in the image file as a data URL.
```
Comment (maurodoglio) [inline atmo/static/js/keys.js]: s/image/ssh key

File: atmo/static/js/keys.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      })
+    }
+
+    var handleFileDrop = function(event) {
+      event.stopPropagation();
+      event.preventDefault();
+
+      var files = event.originalEvent.dataTransfer.files; // FileList object.
+
+      // Loop through the FileList and render image files as thumbnails.
```
Comment (maurodoglio) [inline atmo/static/js/keys.js]: please remove this comment

Comment (jezdez) [inline atmo/static/js/keys.js]: Yikes, c/p strikes again

File: atmo/keys/views.py (hunk 2 above)
Comment (jezdez) [inline atmo/keys/views.py]: woops, forgot to remove this

**Pull request merged**